runhouse/resources/functions/aws_lambda.py

```diff
@@ -0,0 +1,845 @@
+import base64
+import contextlib
+import copy
+import inspect
+import json
+import logging
+import os
+import time
+import warnings
+import zipfile
+from pathlib import Path
+from typing import Any, Callable, List, Optional
+
+import boto3
+import botocore.exceptions
+
+from runhouse.globals import rns_client
+from runhouse.resources.envs import _get_env_from, Env
+
+from runhouse.resources.functions.function import Function
+
+logger = logging.getLogger(__name__)
+
+CRED_PATH = f"{Path.home()}/.aws/credentials"
+SUPPORTED_RUNTIMES = [
+    "python3.7",
+    "python3.8",
+    "python3.9",
+    "python3.10",
+    "python 3.11",
+]
+DEFAULT_PY_VERSION = "python3.9"
+
+
+class AWSLambdaFunction(Function):
+    RESOURCE_TYPE = "lambda_function"
+    DEFAULT_ACCESS = "write"
+    DEFAULT_ROLE_POLICIES = [
+        "cloudwatch:*",
+        "lambda:Invoke",
+        "lambda:InvokeAsync",
+        "lambda:InvokeFunction",
+        "lambda:PublishVersion",
+        "logs:*",
+        "s3:DeleteObject",
+        "s3:GetObject",
+        "s3:ListBucket",
+        "s3:PutObject",
+    ]
+    MAX_WAIT_TIME = 60  # seconds, max time that can pass before we raise an exception that AWS update takes too long.
+    DEFAULT_REGION = "us-east-1"
+    DEFAULT_RETENTION = 30  # one month, for lambdas log groups.
+    DEFAULT_TIMEOUT = 900  # sec. meaning 15 min.
+    DEFAULT_MEMORY_SIZE = 1024  # MB
+    DEFAULT_TMP_SIZE = 3072  # MB, meaning 3G
+
+    # setup needs to be here and not in __init__ for the reload option.
+    if Path(CRED_PATH).is_file():
```

Comment:
Doesn't this mean that whenever we reload a func it's region will always be the default? what if instead of relying on the `LAMBDA_CLIENT` for reloading we used Den?